src/odb/src/db/dbModInst.cpp

```diff
@@ -421,13 +421,22 @@ bool dbModInst::swapMaster(dbModule* new_module)
   const char* old_module_name = old_module->getName();
   const char* new_module_name = new_module->getName();
 
+  // Check if module names differ
+  if (strcmp(old_module_name, new_module_name) == 0) {
+    logger->warn(utl::ODB,
+                 470,
+                 "The modules cannot be swapped because the new module {} is "
+                 "identical to the existing module",
+                 new_module_name);
+  }
```

Comment:
Should there be a return here if the swap can't continue or use error()?

Reply:
Good catch. Fixed.